src/mscorlib/shared/System/Collections/HashHelpers.cs

```diff
@@ -33,6 +33,133 @@ internal static class HashHelpers
+            new PrimeInfo(7013, 0x9584d635, 12),
+            new PrimeInfo(8419, 0x7c8c7b75, 12),
+            new PrimeInfo(10103, 0x33e4f01d, 11),
+            new PrimeInfo(12143, 0x565a3073, 12),
+            new PrimeInfo(14591, 0x23eeaa5d, 11),
+            new PrimeInfo(17519, 0x77b510e9, 13),
+            new PrimeInfo(21023, 0x63c14fe5, 13),
+            new PrimeInfo(25229, 0x531fe999, 13),
+            new PrimeInfo(30293, 0x8a75366b, 14),
+            new PrimeInfo(36353, 0xe6c11447, 15),
+            new PrimeInfo(43627, 0xc047bac3, 15),
+            new PrimeInfo(52361, 0x0a035099, 11),
+            new PrimeInfo(62851, 0x42bbed05, 14),
+            new PrimeInfo(75431, 0x379ac159, 14),
+            new PrimeInfo(90523, 0x05cab127, 11),
+            new PrimeInfo(108631, 0x9a713743, 16),
+            new PrimeInfo(130363, 0x80b236c9, 16),
+            new PrimeInfo(156437, 0x6b3eeec1, 16),
+            new PrimeInfo(187751, 0xb2b7bcf9, 17),
+            new PrimeInfo(225307, 0x4a76bbc7, 16),
+            new PrimeInfo(270371, 0x7c1aeabf, 17),
+            new PrimeInfo(324449, 0x676b743d, 17),
+            new PrimeInfo(389357, 0x2b16ec6d, 16),
+            new PrimeInfo(467237, 0x8fa1117f, 18),
+            new PrimeInfo(560689, 0x77b0a38f, 18),
+            new PrimeInfo(672827, 0x63bddbb1, 18),
+            new PrimeInfo(807403, 0x0531def9, 14),
+            new PrimeInfo(968897, 0x8a86bc61, 19),
+            new PrimeInfo(1162687, 0x737002ad, 19),
+            new PrimeInfo(1395263, 0x180c7f9f, 17),
+            new PrimeInfo(1674319, 0x140a67af, 17),
+            new PrimeInfo(2009191, 0x42cd47bf, 19),
+            new PrimeInfo(2411033, 0x37ab0b5f, 19),
+            new PrimeInfo(2893249, 0x5cc7a9dd, 20),
+            new PrimeInfo(3471899, 0x4d510d43, 20),
+            new PrimeInfo(4166287, 0x080dc5ad, 17),
+            new PrimeInfo(4999559, 0x035b11b9, 16),
+            new PrimeInfo(5999471, 0xb2f90627, 22),
+            new PrimeInfo(7199369, 0x9524d54d, 22),
+            new PrimeInfo(14398753, 0x4a92658f, 22),
+            new PrimeInfo(28797523, 0x4a9262ad, 23),
+            new PrimeInfo(57595063, 0x9524c277, 25),
+            new PrimeInfo(115190149, 0x9524c083, 26),
+            new PrimeInfo(230380307, 0x4a926011, 26),
+            new PrimeInfo(460760623, 0x9524bff1, 28),
+            new PrimeInfo(921521257, 0x9524bfd3, 29),
+            new PrimeInfo(1843042529, 0x4a925fdf, 29),
+            new PrimeInfo(2146435069, 0x20040081, 28)
+        };
+
+        // To implement magic-number divide with a 32-bit magic number,
+        // multiply by the magic number, take the top 64 bits, and shift that
+        // by the amount given in the table.
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static uint MagicNumberDivide(uint numerator, uint magic, int shift)
+        {
+            return (uint)((numerator * (ulong)magic) >> (32 + shift));
```

Comment:
Looks like `+ 32` can be folded into the magic table.

Reply:
We'll need to check if this works well on 32 bit targets. I think RyuJIT x86 handles these particular `long` operations well but I do not know what happens on ARM32.